### PR TITLE
[angular] Fix typo and simplify + add tests to item count component

### PR DESCRIPTION
**Pull request by kaidohallik**
---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/angular/src/main/webapp/app/shared/pagination/item-count.component.ts.ejs [hunk 1]
```diff
@@ -40,7 +40,7 @@ export class ItemCountComponent {
    */
   @Input() set params(params: { page?: number; totalItems?: number; itemsPerPage?: number }) {
     if (params.page !== undefined && params.totalItems !== undefined && params.itemsPerPage !== undefined) {
-      this.first = (params.page - 1) * params.itemsPerPage === 0 ? 1 : (params.page - 1) * params.itemsPerPage + 1;
+      this.first = (params.page - 1) * params.itemsPerPage + 1;
       this.second = params.page * params.itemsPerPage < params.totalItems ? params.page * params.itemsPerPage : params.totalItems;
     } else {
```
Comment (mshima) [inline generators/client/templates/angular/src/main/webapp/app/shared/pagination/item-count.component.ts.ejs:45]: Is this correct for `page: 0`, `totalItems: 0`, `itemsPerPage: 0`?
Is it supposed to happen?

Comment (kaidohallik) [reply generators/client/templates/angular/src/main/webapp/app/shared/pagination/item-count.component.ts.ejs:45]: This component input parameters can have initial values in the parent component. So yes, probably can have also those values.
In the forms generated by generator there are `*ngIf` attributes which are hiding this component if no data is loaded yet.

Comment (mshima) [reply generators/client/templates/angular/src/main/webapp/app/shared/pagination/item-count.component.ts.ejs:45]: this.first will be negative for `page: 0`.
And having `itemsPerPage: 0` the item number is not calculable.

IMO
The condition should be changed to 
`if (params.page && params.totalItems !== undefined && params.itemsPerPage) {`

Comment (kaidohallik) [reply generators/client/templates/angular/src/main/webapp/app/shared/pagination/item-count.component.ts.ejs]: Done.

**Review by mshima — APPROVED** [review]
LGTM

**Pull request merged**